spec.bs

```diff
@@ -3711,6 +3712,101 @@ RAPPOR</a> noises each coordinate of the bit vector independently, and it is par
   1. Return |histogram|.
 </div>
 
+## Get storage interest groups for owner ## {#get-storage-interest-groups-for-owner-header}
+
+*This first introductory paragraph is non-normative.*
+
+The [=get storage interest groups for owner=] algorithm returns a Web IDL array of interest groups. This allows a Protected Audience buyer within a shared storage worklet to gain insights into their users and to send Private Aggregation reports.
+
+<xmp class="idl">
+dictionary StorageInterestGroup : AuctionAdInterestGroup {
+  unsigned long long joinCount;
+  unsigned long long bidCount;
+  sequence<PreviousWin> prevWinsMs;
+  USVString joiningOrigin;
+  long long timeSinceGroupJoinedMs;
+  long long lifetimeRemainingMs;
+  long long timeSinceLastUpdateMs;
+  long long timeUntilNextUpdateMs;
+  unsigned long long estimatedSize;
+};
+</xmp>
+
+<div algorithm>

```

Comment:
I added null checks when applicable. For null URL fields, their [serialization](https://pr-preview.s3.amazonaws.com/xyaoinum/turtledove/pull/1299.html#serialize-a-url) is automatically `undefined`, so it seems I could skip the null check.

Regarding the numeric and enum fields, the current data storage design doesn't distinguish between a default value and an unset value. I think it should be fine (and is probably only feasible) to return whatever the data storage has. Otherwise, we'd need to redesign the storage in the spec and in the implementation.

Reply:
Resolving, given no further comment.